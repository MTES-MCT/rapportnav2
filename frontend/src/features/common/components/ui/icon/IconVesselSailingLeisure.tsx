import Icon from '../../../../../assets/images/navire_plaisance_loisir.svg'
function IconVesselSailingLeisure() {
  return <img src={Icon} alt="Navire de plaisance de loisir" height={36} />
}

export default IconVesselSailingLeisure
