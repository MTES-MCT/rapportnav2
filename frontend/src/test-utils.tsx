import React, { ReactElement } from 'react'
import { render, renderHook, RenderOptions } from '@testing-library/react'
import UIThemeWrapper from './ui/ui-theme-wrapper'
import { BrowserRouter } from 'react-router-dom'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <UIThemeWrapper>
      <BrowserRouter>{children}</BrowserRouter>
    </UIThemeWrapper>
  )
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllTheProviders, ...options })

const customRenderHook = (hook: any, options?: Omit<RenderOptions, 'wrapper'>) =>
  renderHook(() => hook(), { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
export { customRenderHook as renderHook }
