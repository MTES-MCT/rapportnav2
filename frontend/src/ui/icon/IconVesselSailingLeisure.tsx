import Icon from '../../assets/images/navire_plaisance_loisir.svg'
function IconVesselSailingLeisure() {
  return <img src={Icon} alt="Navire de commerce" />
}

export default IconVesselSailingLeisure
