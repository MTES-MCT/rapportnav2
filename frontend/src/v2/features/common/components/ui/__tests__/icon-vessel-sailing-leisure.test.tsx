import { render } from '../../../../../../test-utils'
import IconVesselSailingLeisure from '../icon-vessel-sailing-leisure'

describe('IconVesselSailingLeisure', () => {
  it('should match the snapshot', () => {
    const wrapper = render(<IconVesselSailingLeisure />)
    expect(wrapper).toMatchSnapshot()
  })
})
