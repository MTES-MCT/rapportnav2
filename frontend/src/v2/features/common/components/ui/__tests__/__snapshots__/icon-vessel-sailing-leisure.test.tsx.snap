// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`IconVesselSailingLeisure > should match the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <img
        alt="Navire de plaisance loisir"
        src="data:image/svg+xml,%3csvg%20height='36'%20viewBox='0%200%2034%2036'%20width='34'%20xmlns='http://www.w3.org/2000/svg'%20xmlns:xlink='http://www.w3.org/1999/xlink'%3e%3cclipPath%20id='a'%3e%3cpath%20d='m0%200h34v36h-34z'/%3e%3c/clipPath%3e%3cg%20clip-path='url(%23a)'%3e%3cpath%20d='m0%200h34v36h-34z'%20fill='transparent'/%3e%3cpath%20d='m3.395%2030.752h32.21a.9.9%200%200%201%20.716%201.435l-3.758%205.021a1.788%201.788%200%200%201%20-1.432.717h-23.263a1.788%201.788%200%200%201%20-1.432-.717l-3.757-5.021a.9.9%200%200%201%20.716-1.435zm7.446-7.173h10.448v-13.088zm14.027-20.758v22.551a1.791%201.791%200%200%201%20-1.789%201.793h-17.823a.9.9%200%200%201%20-.7-1.456l18.72-23.448a.894.894%200%200%201%201.593.56z'%20fill='%233b4559'%20transform='translate(-2.5%20-1.926)'/%3e%3c/g%3e%3c/svg%3e"
      />
    </div>
    <div
      class="rs-toast-provider"
    >
      <div
        class="rs-toast-container rs-toast-container-top-center"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-center"
      />
      <div
        class="rs-toast-container rs-toast-container-top-start"
      />
      <div
        class="rs-toast-container rs-toast-container-top-end"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-start"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-end"
      />
    </div>
  </body>,
  "container": <div>
    <img
      alt="Navire de plaisance loisir"
      src="data:image/svg+xml,%3csvg%20height='36'%20viewBox='0%200%2034%2036'%20width='34'%20xmlns='http://www.w3.org/2000/svg'%20xmlns:xlink='http://www.w3.org/1999/xlink'%3e%3cclipPath%20id='a'%3e%3cpath%20d='m0%200h34v36h-34z'/%3e%3c/clipPath%3e%3cg%20clip-path='url(%23a)'%3e%3cpath%20d='m0%200h34v36h-34z'%20fill='transparent'/%3e%3cpath%20d='m3.395%2030.752h32.21a.9.9%200%200%201%20.716%201.435l-3.758%205.021a1.788%201.788%200%200%201%20-1.432.717h-23.263a1.788%201.788%200%200%201%20-1.432-.717l-3.757-5.021a.9.9%200%200%201%20.716-1.435zm7.446-7.173h10.448v-13.088zm14.027-20.758v22.551a1.791%201.791%200%200%201%20-1.789%201.793h-17.823a.9.9%200%200%201%20-.7-1.456l18.72-23.448a.894.894%200%200%201%201.593.56z'%20fill='%233b4559'%20transform='translate(-2.5%20-1.926)'/%3e%3c/g%3e%3c/svg%3e"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
